File: Transaction.py
import uuid
from enum import Enum
from datetime import datetime


class TransactionType(Enum):
    DEPOSIT = "Deposit"
    WITHDRAW = "Withdraw"
    TRANSFER = "Transfer"


class Transaction:
    def __init__(self, amount, transaction_type, destination):
        self.id = str(uuid.uuid4())
        self.amount = amount
        self.type = transaction_type
        self.destination = destination
        self.transaction_date = datetime.now()

File: Transaction.cs
using System;

namespace WalletApp
{
    public class Transaction
    {
        public string Id { get; }
        public double Amount { get; }
        public TransactionType Type { get; }
        public string Destination { get; }
        public DateTime TransactionDate { get; }

        public enum TransactionType
        {
            Deposit,
            Withdraw,
            Transfer
        }

        public Transaction(double amount, TransactionType type, string destination)
        {
            Id = Guid.NewGuid().ToString();
            Amount = amount;
            Type = type;
            Destination = destination;
            TransactionDate = DateTime.Now;
        }
    }
}
